### ISSUE TITLE
ini_managed can no longer handle existing sections

### ISSUE THREAD

**Issue by jmlowe** [saltstack/salt#42809]
### Description of Issue/Question
If a section of an ini exists the minion is no longer able to parse and fails.

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)
test.sls:
```
/tmp/api-paste.ini:
  ini.options_present:
    - separator: '='
    - strict: True
    - sections:
        test:
          testkey: 'testval'
          secondoption: 'secondvalue'
        test1:
          testkey1: 'testval121'
```

### Steps to Reproduce Issue

salt testnode state.apply test

testnode:
----------
          ID: /tmp/api-paste.ini
    Function: ini.options_present
      Result: True
     Comment: Changes take effect
     Started: 15:43:29.673300
    Duration: 3.666 ms
     Changes:
              ----------
              test:
                  ----------
                  after:
                      ----------
                      secondoption:
                          secondvalue
                      testkey:
                          testval
                  before:
                      None
              test1:
                  ----------
                  after:
                      ----------
                      testkey1:
                          testval121
                  before:
                      None

Summary for testnode
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time:   3.666 ms

salt testnode state.apply test

testnode:
----------
          ID: /tmp/api-paste.ini
    Function: ini.options_present
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/site-packages/salt/state.py", line 1837, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/site-packages/salt/loader.py", line 1794, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/site-packages/salt/states/ini_manage.py", line 90, in options_present
                  __salt__['ini.set_option'](name, {section_name: section_body}, separator)[section_name])
              KeyError: 'test'
     Started: 15:43:37.749075
    Duration: 4.838 ms
     Changes:

Summary for testnode
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:   4.838 ms
ERROR: Minions returned with non-zero exit code

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)

Salt Version:
           Salt: 2017.7.0

Dependency Versions:
           cffi: 1.9.1
       cherrypy: Not Installed
       dateutil: 2.6.0
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
   python-gnupg: 0.3.8
         PyYAML: 3.12
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-51-generic
         system: Linux
        version: Ubuntu 16.04 xenial

**Comment by gtmanfred** [comment]
I am unable to replicate this issue.

```
[root@a45f277bf7dc salt]# rm -f /tmp/api-paste.ini
[root@a45f277bf7dc salt]# touch /tmp/api-paste.ini
[root@a45f277bf7dc salt]# cat /tmp/api-paste.ini
[root@a45f277bf7dc salt]# salt-call state.apply test
local:
----------
          ID: /tmp/api-paste.ini
    Function: ini.options_present
      Result: True
     Comment: Changes take effect
     Started: 16:38:48.523844
    Duration: 2.239 ms
     Changes:
              ----------
              test:
                  ----------
                  after:
                      ----------
                      secondoption:
                          secondvalue
                      testkey:
                          testval
                  before:
                      None
              test1:
                  ----------
                  after:
                      ----------
                      testkey1:
                          testval121
                  before:
                      None

Summary for local
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time:   2.239 ms
[root@a45f277bf7dc salt]# cat /tmp/api-paste.ini

[test]
testkey = testval
secondoption = secondvalue

[test1]
testkey1 = testval121
```
Can you provide what is in your initial .ini file before running this state?

Thanks,
Daniel

**Comment by jmlowe** [comment]
The problem comes with the second run.  The first time is succeeds because the file is empty, run a second time when you expect changed=0 and that is when the bug is triggered for me.

**Comment by gtmanfred** [comment]
Ahh, ok i have replicated this issue now.

I am going to bisect and see when this was caused.

Thanks for reporting.
Daniel

**Comment by gtmanfred** [comment]
This is a duplicate of https://github.com/saltstack/salt/issues/34245

And is fixed here https://github.com/saltstack/salt/pull/41690

And will be fixed in 2017.7.2

Thanks,
Daniel
